index.js

```diff
@@ -4,15 +4,104 @@
 //We are providing you with the design of a board (in the DOM) for a player1, you have to create the board for the player2 using the id property 'board_player2' -> it is the second list (ul) in your index.html file
 //First ask the players for their names (use propmt)
 //Now each time the turn player clicks on any cell of the opponent's board (you have to verify if the player is clicking the right board) the program needs to verify if there is an opponent's ship in that cell. If it is then the opponent has one less ship
-//We want you to store the data of each player in two Player objects. Each object has to store: name, remaining boats, and their respective board.
+//We want you to store the data of each player in two Player objects. Each object has to 
+//store: name, remaining boats, and their respective board.
 //Each board needs to be initialized randomly with '0' and four '1' wich means the state of the cell. Numbers 1 are representing the 4 positions of the player's ships
 //Also we want you to display the name of the turn player in the tag that has the id 'turn_player'. And if there is a winner  a text with: 'Congratulationes {name_player}!! you win'
 //in the index.html file you are going to find 4 more ids: 'name_player1' , 'name_player2' , 'ships_player1' , 'ships_player2'. We want to see the information of each player in the respective elements
 //As our previous Battleship, the winner is the player that hits the 4 opponent's ships first
 //one more Thing create a 'reset' and a 'new game' buttons as childs of the element with the id 'buttons'. the reset button has to start the game again and the new game create a new game with new players and a new random board.
 
+the_number  = 4
+//this is a function to create empty game board
+function create_game_board(number){
+  var game_board_list= [];
+  for (var i = 0; i < number; i++){
+    dummy_list = []
+     for (var j = 0; j < number; j++){
+       dummy_list.push(0)
+     }
+     game_board_list.push(dummy_list)
+  }
+  return game_board_list
+}
+
+//a function to get total ship amount on game_board
+function get_total_ship_amount(board,number){
+  total_ships = 0
+  for (var i = 0; i < number; i++){
+    
+      for (var j = 0; j < number; j++){
+      total_ships += board[i][j]
+      }
+      
+  }
+  return total_ships
+}
+
+// Generate  Ships with random values and add into a gameboard
+function fill_board_with_ships(game_board, number){
+  dummy_list = game_board
+do
+{
+  dummy_list[Math.floor(Math.random() * number)][Math.floor(Math.random() * number)] = 1
+  
+}while( get_total_ship_amount(dummy_list,the_number) < number);
+return dummy_list
+}
+
+//function to make players shoot
+function shoot(board,x,y){
+  if (board[x][y] == 1){
```

Comment:
Not a big deal, but it's good to get in the habit of using `===` when writing javascript as opposed to just `==`. The `===` operator is more strict regarding the type of the values being compared so it can help you avoid some really weird problems down the road that can become hard to debug